### ISSUE TITLE
Pressing Shift immediately completes hangul syllable in Qt6

### ISSUE THREAD

**Issue by Kagami** [ankitects/anki#1899]
Hi,

With the latest version of Qt6 Anki (2.1.53), there's the following problem in ~~all~~ Qt6 input fields (e.g. search in Browse window):

When you want to type some text in Korean with a letter which requires Shift, Anki immediately pastes that letter to the input field and doesn't let you to finish the syllable.

E.g. you want to enter word "쭉". To enter "ㅉ" you have to press Shift+ㅈ, but when you do that, Anki pastes "ㅉ" and doesn't let you to type it as "쭉".

This doesn't happen in Qt5 version.

System: Windows 11 x64
Layout: Korean keyboard (101 key) Type 1

**Comment by dae** [comment]
Which OS are you using? Can this be reproduced with a qwerty keyboard and a Korean input method?

**Comment by Kagami** [comment]
> Which OS are you using?

Windows 11 x64.

> Can this be reproduced with a qwerty keyboard and a Korean input method?

It's standard Korean layout on Windows. I.e. I'm using a physical keyboard to type Korean letters.
(ㅉ corresponds to Shift+W, ㅜ to N, ㄱ to R on qwerty keyboard.)

Keyboard layout: Korean keyboard (101 key) Type 1
Input method: Microsoft IME

**Comment by dae** [comment]
I am unable to reproduce this on either a Windows 10 or 11 machine - I can type Shift+w, then n, then r to get 쭉. If I type r first and then n, it does as you describe. I assume in your case, you are typing n then r?

**Comment by Kagami** [comment]
> I am unable to reproduce this on either a Windows 10 or 11 machine - I can type Shift+w, then n, then r to get 쭉

That's strange. I rebooted, installed Qt6 Anki again, created another Windows user, added Korean layout, started Anki with fresh profile and I still can't type 쭉.

> I assume in your case, you are typing n then r?

I'm typing Shift+w and it pastes ㅉ in the input field (e.g. in the preferences or browse window) right away. While it should wait and let me to enter more letters to enter it as a single syllable. In Qt6 version it becomes ㅉㅜㄱ instead of 쭉.

I have no problems in Qt6 version with e.g. 죽, because it doesn't require Shift.

**Comment by Kagami** [comment]
> I am unable to reproduce this on either a Windows 10 or 11 machine

Oh, I think I got it.

It **does** happen in input fields like the search field in Browse window or Preferences window.
It **does not** happen in input fields when you edit a card in Browse window fields at the bottom or in the Deck option fields.

Seems like the former is Qt6 toolkit fields and the latter is Web engine (chromium?) fields. You probably tried the latter.

Sorry, I should have been more specific from the start.

**Comment by dae** [comment]
I've reported this upstream: https://bugreports.qt.io/browse/QTBUG-104043

**Comment by Kagami** [comment]
Seems like https://bugreports.qt.io/browse/QTBUG-103999 has workaround:

> Afterwards, as a temporary solution, I was able to avoid the korean combination problem by receiving the Shift event as follows.

```python
  self.text_edit.keyReleaseEvent = self.process_key
def process_key(self, event):
   pass
```

By ignoring the Shift event(?)

**Comment by dae** [comment]
There's a risk of introducing regressions by doing that, and applying it to every editable area of the app is impractical.

**Comment by Kagami** [comment]
They marked it as critical, hope they will fix it soon :)

**Comment by dae** [comment]
They claim it's fixed in Qt 6.3.1, which Anki 2.1.54 uses. Could you confirm that the problem is gone, or still exists?

https://bugreports.qt.io/browse/QTBUG-103999

**Comment by Kagami** [comment]
I can confirm, it's fixed in Anki 2.1.54 Qt6 version.
Thank you!
